### PR TITLE
add clef support to ABC import

### PR DESCRIPTION
**Pull request by rettinghaus**

### CODE REVIEW

**Comment by lpugin** [comment]
Reviewing this I realized that there are some placed where char array access is not safe. See in `readMusicCode` and similar methods. Could you :
* change this to std::string 
* use `.at` accessor
* always make sure you are not out of bounds

**Comment by rettinghaus** [comment]
Now it should be fine

File: include/vrv/ioabc.h [hunk 1]
```diff
@@ -56,7 +56,7 @@ class AbcInput : public FileInputStream {
     void parseABC(std::istream &infile);
 
     // parsing functions
-    int SetBarLine(const char *musicCode, int index);
+    int SetBarLine(std::string musicCode, int index);
```
Comment (lpugin) [inline include/vrv/ioabc.h]: Please change this to `std::string &musicCode` for optimization

File: include/vrv/ioabc.h [hunk 2]
```diff
@@ -75,7 +75,7 @@ class AbcInput : public FileInputStream {
 
     // input functions
     void readInformationField(char dataKey, std::string dataValue);
-    void readMusicCode(const char *musicCode, Section *section);
+    void readMusicCode(std::string musicCode, Section *section);
```
Comment (lpugin) [inline include/vrv/ioabc.h]: Same here

**Pull request merged**